aten/src/ATen/native/sparse/SparseBinaryOpIntersectionCommon.h

```diff
@@ -299,28 +279,25 @@ void _sparse_binary_op_intersection_kernel_impl(
     auto indices_nnz_stride = indices.stride(1);
 
     auto hash = at::empty({probably_coalesced._nnz()},
-        indices.options().dtype(kHash));
+        indices.options().dtype(kLong));
```

Comment:
nit, indices dtype is already kLong:
```suggestion
        indices.options());
```

Reply:
See my comments above.